apps/snitch_core/lib/core/tools/validations.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Snitch.Tools.Validations do
+  @moduledoc """
+  A bunch of data validations for `Ecto.Changeset`.
+
+  ## Note
+
+  All validations in this module are check ONLY the value present under the
+  `:changes` key in the `Ecto.Changeset.t()`.
+
+  The validations are non-strict, and will not complain if the key is not
+  present under `:changes`.
+  """
+
+  import Ecto.Changeset
+
+  @doc """
+  Validates that the amount (of type `Money.t`) under the `key` in `changeset`
+  is non-negative.
+  """
+  @spec validate_amount(Ecto.Changeset.t(), atom) :: Ecto.Changeset.t()
+  def validate_amount(%Ecto.Changeset{valid?: true} = changeset, key) when is_atom(key) do
+    case fetch_change(changeset, key) do
+      {:ok, %Money{amount: amount}} ->
+        if Decimal.cmp(amount, Decimal.new(0)) != :lt do
+          changeset
+        else
+          add_error(changeset, key, "must be greater than 0", validation: :number)
+        end
+
+      :error ->
+        changeset
+    end
+  end
+
+  def validate_amount(changeset, _), do: changeset
+
+  @doc """
+  Validates that the given date (of type `DateTime.t`) under the `key` in
+  `changeset` is in the future wrt. `DateTime.utc_now/0`.
+  """
+  @spec validate_future_date(Ecto.Changeset.t(), atom) :: Ecto.Changeset.t()
+  def validate_future_date(changeset, key) when is_atom(key) do
```

Comment:
Is there a need to pattern match changset `is_valid` here ?

Reply:
yeah missed this updating.